core/string_api.php

```diff
@@ -721,75 +718,53 @@ function string_get_bug_view_url_with_fqdn( $p_bug_id ) {
- * @param integer $p_user_id A valid user identifier.
- * @return string
- */
-function string_get_bug_view_page( $p_user_id = null ) {
-	return string_get_bug_page( 'view' );
-}
-
-/**
  * return an href anchor that links to a bug UPDATE page for the given bug
- * account for the user preference and site override
  * @param integer $p_bug_id  A bug identifier.
- * @param integer $p_user_id A valid user identifier.
  * @return string
  */
-function string_get_bug_update_link( $p_bug_id, $p_user_id = null ) {
+function string_get_bug_update_link( $p_bug_id ) {
 	$t_summary = string_attribute( bug_get_field( $p_bug_id, 'summary' ) );
-	return '<a href="' . helper_mantis_url( string_get_bug_update_url( $p_bug_id, $p_user_id ) ) . '" title="' . $t_summary . '">' . bug_format_id( $p_bug_id ) . '</a>';
+	return '<a href="' . helper_mantis_url( string_get_bug_update_url( $p_bug_id ) ) . '" title="' . $t_summary . '">' . bug_format_id( $p_bug_id ) . '</a>';
 }
 
 /**
- * return the name and GET parameters of a bug UPDATE page for the given bug
- * account for the user preference and site override
+ * return the name and GET parameters of a bug UPDATE page
  * @param integer $p_bug_id  A bug identifier.
- * @param integer $p_user_id A valid user identifier.
  * @return string
  */
-function string_get_bug_update_url( $p_bug_id, $p_user_id = null ) {
-	return string_get_bug_update_page( $p_user_id ) . '?bug_id=' . $p_bug_id;
+function string_get_bug_update_url( $p_bug_id ) {
+	return string_get_bug_update_page() . '?bug_id=' . $p_bug_id;
 }
 
 /**
- * return the name of a bug UPDATE page for the user
- * account for the user preference and site override
- * @param integer $p_user_id A valid user identifier.
+ * return the name of a bug UPDATE page
  * @return string
  */
-function string_get_bug_update_page( $p_user_id = null ) {
+function string_get_bug_update_page() {
 	return string_get_bug_page( 'update' );
 }
 
 /**
- * return an href anchor that links to a bug REPORT page for the given bug
- * account for the user preference and site override
- * @param integer $p_user_id A valid user identifier.
+ * return an href anchor that links to a bug REPORT page
  * @return string
  */
-function string_get_bug_report_link( $p_user_id = null ) {
-	return '<a href="' . helper_mantis_url( string_get_bug_report_url( $p_user_id ) ) . '">' . lang_get( 'report_bug_link' ) . '</a>';
+function string_get_bug_report_link() {
+	return '<a href="' . helper_mantis_url( string_get_bug_report_url() ) . '">' . lang_get( 'report_bug_link' ) . '</a>';
 }
 
 /**
- * return the name and GET parameters of a bug REPORT page for the given bug
- * account for the user preference and site override
- * @param integer $p_user_id A valid user identifier.
+ * return the name of a bug REPORT page
  * @return string
  */
-function string_get_bug_report_url( $p_user_id = null ) {
-	return string_get_bug_report_page( $p_user_id );
+function string_get_bug_report_url() {
```

Comment:
Why do we still need string_get_bug_report_url() and string_get_bug_report_page() rather than just one of them?